content/articles/garbage-collection-in-c#-dotnet-framework/index.md

```diff
@@ -0,0 +1,207 @@
+author: elijah-maina
+date: 2021-10-12T00:00:00-15:26
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url:  /engineering-education/garbage-collection-in-c#-dotnet-framework/hero.jpg
+    alt: Garbage collection in c# dotnet framework image
+---
+
+Garbage collection in the C#.Net framework is used to manage Memory automatically. In runtime, heap storage space can be allocated to an object class. If the software has completed all the activities related to the object, the memory location will be a waste since it will not be usable.
+ <!--more-->
+### Introduction
+Garbage collection is a critical approach in the .Net framework for releasing unused handled code objects from Memory and allowing the process to run faster. In this tutorial, we will go over the fundamentals of garbage collection.
+
+### Table of contents
+- [Overview on Garbage Collection](#overview-on-garbage-collection)
+- [Garbage Collections Different phases](#garbage-collections-different-phases)
+- [Garbage Collection Heap generation](#garbage-collection-heap-generation)
+- [Methods in Garbage Collection Class](#methods-in-garbage-collection-class)
+- [Stand out advantages of Garbage Collection](#stand-out-advantages-of-garbage-collection)
+
+### Overview on garbage collection
+In a .Net model, memory management is usually automated through the help of Garbage collection. A stack storage space is assigned only when a runtime class object is generated. Whenever the software completes all the activities connected with the thing, the Memory linked to it will not be helpful as it will not be utilized.
+
+Memory space that is not required is freed up with the help of garbage collection, which makes it worthwhile to use. Inbuilt scripting engine, the `Optimization Engine` found in the managed heap, is always used in garbage collection.
+When at least one of several conditions is met, garbage collection occurs. 
+
+The following are the circumstances:
+- When the primary storage becomes limited, garbage collection becomes useful.
+- Whenever a stack memory surpasses a certain threshold, garbage collection becomes useful and starts when the amount of Memory differs from each object.
+- It is only when `GarbageC.Collect` is initiated that garbage collection happens. Because the waste collection is typically automated, this method is only employed in exceptional instances.
+
+### How garbage collection works
+The .Net framework must support implicit garbage collection. When an item is formed, it is assigned to Generation 0. Garbage collection employs an algorithm that examines the items in the generation; when the object's life period expires, it is removed from Memory.
+
+There are two types of things; Live Objects and Dead Objects. The Garbage Collection mechanism collects any unutilized objects in the generation that are dead instances.
+
+If a living item has been running for a long time, it will be shifted to the next generation, depending on how long it has been running.

```

Comment:
These are quite short (single line) paragraphs. Please join these up (the three lines) into one bigger paragraph.